### ISSUE TITLE
Undefined reference to singleteon declared in shared library

### ISSUE THREAD

**Issue by d-gurin** [DOCGroup/ACE_TAO#1987]
### Version

ACE_VERSION "7.0.9"
(with pre-generated makefile's)

### Host machine and operating system

```
Linux work-pc 5.14.0-1054-oem #61-Ubuntu SMP Fri Oct 14 13:05:50 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

### Compiler name and version (including patch level)

```
g++ (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
```

### The $ACE_ROOT/ace/config.h file

`config-g++-common.h`

### AREA/CLASS/EXAMPLE AFFECTED:

Macros: `ACE_EXPORT_SINGLETON_DECLARE`

### The problem effects:

Unable to link my application when linking with my shared library.

### Synopsis

Failed to link own application with own shared library having singleton object declared via `ACE_EXPORT_SINGLETON_DECLARE` macro.

### Description

The application consists of 2 modules:
- shared library, which exports a class wrapped in singleton with `ACE_EXPORT_SINGLETON_DECLARE`
- executable importing the singleton via `ACE_IMPORT_SINGLETON_DECLARE`

The application can't be linked with errors like:
```
/usr/bin/ld: ../mylib/libmylib.so: undefined reference to `ACE_TSS_Singleton<MyLibraryClass, ACE_Thread_Mutex>::instance()'
```

Basically, the issue is caused by this commit: https://github.com/DOCGroup/ACE_TAO/commit/7771149a115f4225b32990e97b5dc01766058ca4

After this commit there is `extern` in both IMPORT and EXPORT macro, which prevents the template to be instantiated inside in the application (IMPORT macro - OK) and in the shared library (EXPORT macro - wrong).

### Sample fix/ workaround

Workaround - use static libraries instead of shared libraries.

Fix - reverting the commit https://github.com/DOCGroup/ACE_TAO/commit/7771149a115f4225b32990e97b5dc01766058ca4
